app/modules/asset_groups/models.py

```diff
@@ -432,12 +432,28 @@ def get_encounters_json(self):
 
     @classmethod
     def check_jobs(cls):
-        for asset_group_sighting in AssetGroupSighting.query.all():
-            asset_group_sighting.check_all_job_status()
+        # get scheduled celery tasks only once and use for all AGS
+        from app.utils import get_celery_tasks_scheduled
+
+        all_scheduled = get_celery_tasks_scheduled(
+            'app.modules.asset_groups.tasks.sage_detection'
+        )
+        for asset_group_sighting in AssetGroupSighting.query.filter(
+            AssetGroupSighting.stage == AssetGroupSightingStage.detection
+        ).all():
+            asset_group_sighting.check_all_job_status(all_scheduled)
 
-    def check_all_job_status(self):
+    def check_all_job_status(self, all_scheduled):
         jobs = self.jobs
         if not jobs:
+            if not all_scheduled:
+                # TODO it would be nice to know if the scheduled tasks were for other AGS than this one
+                # but at the moment, it's not clear how we could detect this
```

Comment:
you can grab the `id` from the _async resource_ (i think it is called) when celery kicks off a task.  see this line here, where i am capturing `async_res` and then later i return `async_res.id`.  this is just a guid that is the internal id within celery of the job.  for merge-requests, i tied them to this id, so that i could know which _specific_ celery job i could cancel (e.g. in the event the merge request was approved prior to the celery task reaching its deadline).

you could possibly store this guid (for celery) along with the IA job somewhere?

Reply:
The thing is, what I'm actually looking for here are tasks/jobs that have been created by the Houston container but not yet run by Celery. What I'm looking for are jobs that have not yet been run. 
I may be wrong but what I *think* you're talking about are jobs that have been processed by celery and executed?

Reply:
I'm hoping we can go with this for now, it's to catch the problem when things are not being processed by celery and this will find it. 
We can, in future, finesse this a bit for for now, I'd like to go with this simple solution